pkg/sentry/fsimpl/fuse/dentry.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fuse
+
+import (
+	"sync"
+
+	"gvisor.dev/gvisor/pkg/refs"
+	ktime "gvisor.dev/gvisor/pkg/sentry/kernel/time"
+	"gvisor.dev/gvisor/pkg/sentry/vfs"
+)
+
+// dentry implements vfs.DentryImpl.
+type dentry struct {
+	vfsd vfs.Dentry
+
+	refs.AtomicRefCount
+
+	// the owning filesystem. fs is immutable.
+	fs *filesystem
+
+	// file size.
+	size uint64
+
+	// access, change and modification time of the file
```

Comment:
Should be maintained by the FUSE daemon?

Reply:
In Linux, it is maintained by on inode https://elixir.bootlin.com/linux/v4.4/source/fs/fuse/file.c#L191

Reply:
Yup, if using writeback cache

Reply:
> In Linux, it is maintained by on inode https://elixir.bootlin.com/linux/v4.4/source/fs/fuse/file.c#L191

should we place them on inode too?

We are not going to support the writeback cache right now, so I am not sure that we need to care about this.